### PR TITLE
driver: add UGV platform and boarAI resource descriptor

### LINKED ISSUE #38: implement driver for boarAI (UGV) with mavlink
implement vehicle driver based on mavlink to support boarAI as a resource in the HMI ecosystem.

### PR DESCRIPTION
**Pull request by sergiomiracco**
Closes #38

### CODE REVIEW

File: app/src/main/kotlin/ch/hsr/ifs/gcs/driver/mavlink/platform/CommonUGV.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    enum class UGVCustomMode(val id: Int) {
+        MANUAL(1),
+        SPEED_CONTROL(2),
+        POSITION_CONTROL(3),
+        AUTOMATIC(4),
+        OFFBOARD_CONTROL(5)
+    }
+
+    enum class PX4AutomaticModeSubMode(val id: Int) {
```
Comment (fmorgner) [inline app/src/main/kotlin/ch/hsr/ifs/gcs/driver/mavlink/platform/CommonUGV.kt]: We don't need those. They are an artifact of the old PX4 implementation

File: app/src/main/kotlin/ch/hsr/ifs/gcs/driver/mavlink/platform/CommonUGV.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+         *
+         * @since 1.2.0
+         * @author ILT Institute for Lab Automation and Mechatronics
+         */
+        const val DRIVER_MAVLINK_COMMON_UGV = "ch.hsr.ifs.gcs.driver.mavlink.platform.CommonUGV"
+
+        fun instantiate(channel: Channel, payloads: List<Payload>): CommonUGV? = CommonUGV(channel, payloads)
+    }
+
+    enum class UGVCustomMode(val id: Int) {
```
Comment (fmorgner) [inline app/src/main/kotlin/ch/hsr/ifs/gcs/driver/mavlink/platform/CommonUGV.kt]: Not sure about those. We should be able to facilitate (almost) the same functionality using standard MAV_MODE_FLAG

**Review by fmorgner — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/ch/hsr/ifs/gcs/driver/mavlink/platform/CommonUGV.kt (hunk 1 above)
Comment (sergiomiracco) [reply app/src/main/kotlin/ch/hsr/ifs/gcs/driver/mavlink/platform/CommonUGV.kt]: deleted

File: app/src/main/kotlin/ch/hsr/ifs/gcs/driver/mavlink/platform/CommonUGV.kt (hunk 2 above)
Comment (sergiomiracco) [reply app/src/main/kotlin/ch/hsr/ifs/gcs/driver/mavlink/platform/CommonUGV.kt]: deleted

**Review by fmorgner — APPROVED** [review]

**Pull request merged**